public/app/features/explore/state/reducers.ts

```diff
@@ -687,13 +688,17 @@ const getModesForDatasource = (dataSource: DataSourceApi, currentMode: ExploreMo
     supportedModes.push(ExploreMode.Logs);
   }
 
+  if (supportsTracing) {
+    supportedModes.push(ExploreMode.Tracing);
+  }
+
   if (supportedModes.length === 1) {
     mode = supportedModes[0];
   }
 
   // HACK: Used to set Loki's default explore mode to Logs mode.
   // A better solution would be to introduce a "default" or "preferred" mode to the datasource config
-  if (dataSource.meta.name === 'Loki' && !currentMode) {
+  if (dataSource.meta.name === 'Loki' && (!currentMode || supportedModes.indexOf(currentMode) === -1)) {
```

Comment:
```suggestion
  if (dataSource.meta.name === 'Loki' && (!currentMode || !supportedModes.includes(currentMode))) {
```

Personal preference but I think this reads a bit nicer